plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/StopTask.java

```diff
@@ -0,0 +1,27 @@
+package org.wildfly.swarm.plugin.gradle;
+
+import java.util.concurrent.TimeUnit;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.TaskAction;
+import org.wildfly.swarm.tools.exec.SwarmProcess;
+
+public class StopTask extends DefaultTask {
+
+    @TaskAction
+    public void stopThorntailProcess() {
+        SwarmProcess process = (SwarmProcess) getProject().findProperty(PackagePlugin.THORNTAIL_PROCESS_PROPERTY);
```

Comment:
As discussed in the above comment, we might need to look at the scenario where the process was kicked off and is running in the background. The Gradle daemon might have crashed and now we wish to stop the process. In such a scenario, if the process ID is available in the `buildDir`, then we can attempt to stop the process (if it is still running). If there is no known processID then we just ignore.

Reply:
One thing that discouraged me from this was that there is no cross-platform way of doing it. There's no API to even retrieve process ID until JDK 9, and then there would be the killing part...

Reply:
IMO we should postpone this until newer JDK. With JDK 8 it would require us to use reflection to access private field on JDK type (`Process#pid`), which would trigger illegal access warnings on newer JDKs. It doesn't feel right to add these kind of reflections now, when we should rather be actively removing them. WDYT?

Reply:
Agreed